viscy/data/dynacell.py

```diff
@@ -0,0 +1,224 @@
+"""Test stage data modules for loading data from DynaCell benchmark datasets."""
+
+import logging
+from pathlib import Path
+from typing import Sequence
+
+import pandas as pd
+import torch
+from iohub.ngff import open_ome_zarr
+from lightning.pytorch import LightningDataModule
+from torch.utils.data import ConcatDataset, DataLoader
+
+from viscy.data.segmentation import TargetPredictionDataset
+from viscy.data.typing import DynaCellSample
+
+_logger = logging.getLogger("lightning.pytorch")
+
+
+class DynaCellDataset(TargetPredictionDataset):
+    """Return a DynaCellSample object with the cell type, organelle, and infection condition."""
+
+    def __init__(
+        self,
+        cell_type: str,
+        organelle: str,
+        infection_condition: str,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.cell_type = cell_type
+        self.organelle = organelle
+        self.infection_condition = infection_condition
+
+    def __getitem__(self, idx) -> DynaCellSample:
+        sample = super().__getitem__(idx)
+
+        # Convert tensors to float32 for metrics compatibility
+        sample["pred"] = sample["pred"].float()
+        sample["target"] = sample["target"].float()
+
+        # Add channel dimension if needed for the metrics (BxHxW -> BxCxHxW)
+        if sample["pred"].ndim == 2:
+            sample["pred"] = sample["pred"].unsqueeze(0)  # Add channel dimension
+        elif sample["pred"].ndim == 3 and sample["pred"].shape[0] == 1:
+            # If the first dimension is batch with size 1, reshape to [C,H,W]
+            sample["pred"] = sample["pred"].squeeze(0).unsqueeze(0)
+
+        if sample["target"].ndim == 2:
+            sample["target"] = sample["target"].unsqueeze(0)  # Add channel dimension
+        elif sample["target"].ndim == 3 and sample["target"].shape[0] == 1:
+            # If the first dimension is batch with size 1, reshape to [C,H,W]
+            sample["target"] = sample["target"].squeeze(0).unsqueeze(0)
+
+        sample.update(
+            {
+                "cell_type": self.cell_type,
+                "organelle": self.organelle,
+                "infection_condition": self.infection_condition,
+            }
+        )
+        return sample
+
+
+class DynaCellDataBase:
```

Comment:
Capital `XxxBase` reads like the name of a base class. Also this is not a databse with a runtime, but a dataframe normalizer.